lib/index.js

```diff
@@ -0,0 +1,255 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const assert = require('assert-plus');
+const base64url = require('base64url-universal');
+const bedrock = require('bedrock');
+const crypto = require('crypto');
+const database = require('bedrock-mongodb');
+const {promisify} = require('util');
+const uuid = require('uuid-random');
+const {util: {BedrockError}} = bedrock;
+
+// load config defaults
+require('./config');
+
+const AES_KW_ALGORITHM = 'id-aes256-wrap';
+const AES_KW_RFC3394_IV = new Buffer.from('A6A6A6A6A6A6A6A6', 'hex');
+
+bedrock.events.on('bedrock-mongodb.ready', async () => {
+  await promisify(database.openCollections)(['ssm']);
+
+  await promisify(database.createIndexes)([{
+    // cover queries by ID
+    collection: 'ssm',
+    fields: {id: 1},
+    options: {unique: true, background: false}
+  }, {
+    // cover queries by controller
+    collection: 'ssm',
+    fields: {controller: 1},
+    options: {unique: false, background: false}
+  }]);
+});
+
+/**
+ * @module bedrock-ssm-mongodb
+ */
+
+/**
+ * Generates a new key.
+ *
+ * @param {Object} options - The options to use.
+ * @param {string} options.controller - The ID of the controller of the key.
+ * @param {string} options.type - The type of key (e.g. 'AES-KW', 'HS256').
+ *
+ * @returns {Promise<Object>} An object containing `{id}`.
+ */
+exports.generateKey = async ({controller, type}) => {
+  assert.string(controller, 'options.controller');
+  assert.string(type, 'options.type');
+
+  let key;
+  const id = uuid();
```

Comment:
I see that key ids here are UUIDs, is it necessary to hash before storing?  And I notice that the id is not being stored in the body of the key `key.id` per our normal practice.  Is that a feature or a bug?

Reply:
I'm still not sure what we want to do with key IDs. I need to sit down and have a design conversation with interested parties about it. I don't mind if we push forward with what we have right now -- provided that we get the design issues sorted out in the very near future.